include/zephyr/bluetooth/gatt.h

```diff
@@ -1660,6 +1672,9 @@ struct bt_gatt_subscribe_params {
 	/** Notification value callback */
 	bt_gatt_notify_func_t notify;
 	/** Subscribe CCC write request response callback */
+	bt_gatt_subscribe_func_t subscribe;
+
+	/** DEPRECATED subscribe CCC write response callback */
```

Comment:
Since we cannot properly deprecate it, is this the right way to do it @carlescufi ? Shouldn't we just remove the callback?

Reply:
Yes, a stable API change means that, changing a stable API. So if we are not changing the existing `write` callback then we should remove it from here.

Reply:
@asbjornsabo why are we not changing the signature of `bt_gatt_write_func_t` instead? is it because it's used in other places?

Reply:
Understood, thank you @asbjornsabo.
@Thalley, since you added that callback, would you mind ACKing this change here?

Reply:
I am in favor of replacing the `write` callback with the new callback completely (i.e. remove `write` and add `subscribe`, without deprecation).

Reply:
Dsicussed in discord - keep deprecated callback, as that is cheap.